echopype/echodata/combine.py

```diff
@@ -435,14 +446,296 @@ def _check_echodata_channels(
+        for ds in ds_list:
+            times = ds[time].values
+            if isinstance(times, np.ndarray):
+                # store first time if we have an array
+                first_times.append(times[0])
+            else:
+                # store first time if we have a single value
+                first_times.append(times)
+
+        first_times = np.array(first_times)
+
+        # skip check if all first times are NaT
+        if not np.isnan(first_times).all():
+            is_descending = (np.diff(first_times) < np.timedelta64(0, "ns")).any()
+
+            if is_descending:
+                raise RuntimeError(
+                    f"The coordinate {time} is not in ascending order for "
+                    f"group {ed_group}, combine cannot be used!"
+                )
+
+
+def _check_filter_params(
+    ds_list: List[xr.Dataset], ed_group: Literal["Vendor_specific"], ds_append_dims: set
+) -> None:
+    """
+    Check for identical filter params for all inputs in Vendor specific group
+
+    Parameters
+    ----------
+    ds_list: list of xr.Dataset
+        List of Datasets to be combined
+    ed_group: "Vendor_specific"
+        The name of the ``EchoData`` group being combined,
+        this is only works for "Vendor_specific" group.
+    ds_append_dims: set
+        A set of datasets append dimensions
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    ValueError
+        If ``ed_group`` is not ``Vendor_specific``.
+    RuntimeError
+        If non identical filter parameters is found.
+    """
+    if ed_group != "Vendor_specific":
+        raise ValueError("Group must be `Vendor_specific`!")
+
+    if len(ds_append_dims) > 0:
+        # If there's a dataset appending dimension, drop for comparison
+        # of the other values... everything else should be identical
+        ds_list = [ds.drop_dims(ds_append_dims) for ds in ds_list]
+
+    it = iter(ds_list)
+    # Init as identical, must stay True.
+    is_identical = True
+    dataset = next(it)
+    for next_dataset in it:
+        is_identical = dataset.identical(next_dataset)
+        if not is_identical:
+            raise RuntimeError(
+                f"Non identical filter parameters in {ed_group} group. " "Objects cannot be merged!"
+            )
+        dataset = next_dataset
+
+
+def _merge_attributes(attributes: List[Dict[str, str]]) -> Dict[str, str]:
+    """
+    Merges a list of attributes dictionary
+
+    Parameters
+    ----------
+    attributes : list of dict
+        List of attributes dictionary
+        E.g. [{'attr1': 'val1'}, {'attr2': 'val2'}, ...]
+
+    Returns
+    -------
+    dict
+        The merged attribute dictionary
+    """
+    merged_dict = {}
+    for attribute in attributes:
+        for key, value in attribute.items():
+            if value == "" and key not in merged_dict:
+                merged_dict[key] = value
+            elif value != "":
+                merged_dict[key] = value
```

Comment:
This logic looks like if `key` already exists in `merged_dict` from a previous `attribute` in the loop, it will be overwritten with the latest `value`, but I think we want to do the reverse -- ie to keep only the first `value` entry?

Probably would be a good idea to add some comments to the if-else block to make it obvious what the expected handling is for the new key-value pair

Reply:
I think it would be better to keep the latest because if a user wants to append a new data, rather than messing with the already combined data, it might be better to update the attribute in the new data, which will overwrite the current attributes in the first data to be appended. This was the reasoning behind that. If I remember correctly, I had a discussion about this with @emiliom and he thought that was good. Could you chime in @emiliom  and let me know what you think? Thanks 😄

Reply:
I think one of the attribute is the first ping time, so using the latest one doesn't make sense...

Reply:
Which one is that for?

Reply:
i think it's in the AZFP vendor group -- in a sense it is not needed, since it is already in the dataset. 

In a sense using neither the first nor the last makes sense. Let's discuss in the slack channel...

Reply:
In the Slack channel you asked that I comment here. I don't think I follow the code logic, since both parts of the if statement result in `merged_dict[key] = value`. So, I'll comment on the discussion. I'm not sure what @leewujung means by "I think one of the attributes is the first ping time, so using the latest one doesn't make sense...", but in my discussion with @lsetiawan we talked about storing either the first or last values. Based on that discussion alone, there weren't clear arguments for picking one over the other (the first or the last value). So, if you have good arguments for one, that's fine with me.